### DISCUSSION TITLE
How to use `nx run-many` with `--exclude` flag?

### DISCUSSION THREAD

**Discussion by mcmxcdev** [nrwl/nx#10942] [General]
We have a monorepo that contains two next.js applications. Since `next build` already does typechecking, we want to skip that step for the actual `typecheck` executor.

I tried to use `nx run-many --target=typecheck --all --exclude='[A,B]'`, but it still runs all projects due to the `--all` flag.
This is the relevant docs that I referred to: https://nx.dev/cli/run-many#exclude

Specifying projects directly with `--projects=` is not really useful here, as we want this to scale whenever we add new packages to the monorepo.

**Comment by ivandjomegni** [comment]
hi @mcmxcdev I use nx run-many  with the --exclude option and it appears to work for me for the most part, even though I am still trying to figure out other issues I have, but your issue seems to be on your syntax, try to remove the quotes and do 
`nx run-many --target=typecheck --all --exclude=[A,B]`

With this one though I found that sometimes it still adds certain projects specified in the excluded options, could be a cache on my end but this syntax should work.

**Reply by mcmxcdev** [reply]
Thank you Ivan for getting back at me!

Without quotes, I am running into `zsh: no matches found: --exclude=[A,B]`

Luckily I am not dependent on this functionality any more right now.

**Comment by AgentEnder** [comment]
I think this is the command you are looking for: `nx run-many --target=typecheck --all --exclude=A,B`

**Reply by mcmxcdev** [reply]
That does the trick, thanks!

**Reply by cmbkla** [reply]
I ended up on this thread, because the documentation here says the argument for projects is a "string", but for exclude it says "array", so I am over here trying to figure out what format it wants the array in. This appears to be a documentation flaw. https://nx.dev/nx/run-many

**Reply by veloware** [reply]
the exact same reason I'm here too...

**Comment by Mills-2** [comment]
nx seems to ignore the `--exclude` flag for me. Running `nx run-many --target=build --exclude=A` runs both A and B.

**Reply by AgentEnder** [reply]
Is A perhaps a dependency of B? `--exclude` doesn't affect `dependsOn`, so that could explain it still running.

**Reply by Mills-2** [reply]
This appears to be the case, thanks!

**Comment by devzom** [comment]
For me smth like this works in NX 16.x
`nx run-many --target=test --silent --exclude='app-1,app-2'`
